content: Add some human missions about skydiving/spacediving. (#5781)
endless-sky/endless-sky: data/human/transport missions.txt

			label reject
			`	Charles looks surprised for a moment, and then, for the first time you've seen, he smiles. "Thank you, Captain <last>. I wish there were more people like you in the Paradise Worlds."`
			`	You wish the couple the best of luck on Earth, and they slowly make their way to the train station. Presumably, their new apartment is far from here.`

mission "Spacediving professionals"
	name "Spacediving at <destination>"

> 	description "Drop skydivers from the edge of space. They are carrying cooled spacesuits, durable parachutes and oxygen bottles to ensure the survival of their party. Payment is <payment>."

Reviewer: ```suggestion
	description "Head to destination with <bunks> skydivers, then drop them from the edge of space as you land. They are wearing thermal spacesuits, durable parachutes and oxygen bottles to ensure their survival. Payment is <payment>."
```

	passengers 5 10
	repeat

> 	job

Reviewer: These should be at the start of the mission definition.

Author: I don't mind to change the order of the keywords, but can you give me a reference on where the order is described? The example on the Wiki https://github.com/endless-sky/endless-sky/wiki/CreatingMissions seems to use the same order as that I used, so if that order is not correct then we should also update the Wiki.

	source

> 		government "Republic" "Free Worlds" "Syndicate"

Reviewer: ```suggestion
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"
```

	destination

> 		distance 7 9

Reviewer: Couldn't this be a shorter distance?

Author: Yes, changed to 1 .. 7.

> 		government "Republic" "Free Worlds" "Syndicate"

Reviewer: These missions probably need to check whether their destination has enough atmosphere to parachute on, or if their destination is a planet at all, and not a station.
```suggestion
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"
```

Author: Agreed, need to add checks if the destination is suitable for parachuting.

Author: I added two checks for this.

	to offer

> 		and
> 			credits >= 1000000
> 			random > 96

Reviewer: Why does the player need 1 million credits to have this offered?
```suggestion
		credits >= 1000000
		random < 3
```

Author: I reduced the credits requirement and only placed it on the job with the fine.

	on visit
		dialog phrase "generic passenger on visit"
	on complete

> 		payment 8000

Reviewer: Only 8000?

Author: I reduced the distance. What amount would you feel is appropriate for this type of job?

Reviewer 2: Typically payment is at least partially based on distance. Is there any reason to deliberately deviate from that convention?

Author: Allowing people to skydive from your ship carries higher risks than just transporting people from one location to another. I was thinking of just a fixed amount of higher payment regardless of the distance.
(And another reason is that I needed to hardcode the payment value for missions where the payment is actually not paid out.)

Reviewer 3: The payment calculation is as follows: `base + multiplier * (jumps + 1) * (10 * (# of passengers) + (tons of cargo))`.
Given the max possible distance of 7 jumps, the max possible number of passengers of 10, and using a base of 0 and default multiplier of 150, we're looking at `0 + 150 * (7 + 1) * (10 * 10 + 0) = 120k` if this wasn't a fixed payment job. So just a fixed payment of 8k is hardly worth it given what a similarly sized normal passenger mission would offer.

```
mission "Skydiving Test"
	job
	repeat
	description "Head to <destination> with <bunks> passengers for <payment>."
	passengers 10
	destination
		distance 7
	on complete
		payment
```
![image](https://user-images.githubusercontent.com/17688683/123171507-c5623b00-d430-11eb-892c-a12abc07f165.png)

Even a bare minimum of a 1 distance destination with only 5 passengers would result in 15k credits. 
So this payment should definitely be increased. Given that some of the time that you take this mission you won't receive any pay or a reduced pay, I think a payment closer to the maximum possible under normal circumstances would be appropriate, possible even above the maximum.

Author: Agreed. I increased the payment to the maximum calculated here (and adjusted the fine to match).

> 		dialog "The skydivers jump from your ship and make a spectacular dive until they open their parachutes and land safely near the spaceport. You received payment of <payment> just before they jumped out."

Reviewer: ```suggestion
		dialog "As you descend through the upper atmosphere, the skydivers pay you <payment> before jumping from your ship, spectacularly diving until they open their parachutes and land safely near the spaceport."
```

mission "Spacediving amateurs"
	name "Spacediving at <destination>"

> 	description "Drop skydivers from the edge of space. They are carrying cool wingsuits, fashionable parachutes and beer bottles to ensure the trip will be a great party. Payment is 8000 credits."

Reviewer: ```suggestion
	description "Head to destination with <bunks> skydivers, then drop them from the edge of space as you land. They are wearing tricked-out wingsuits, brightly-patterned parachutes and carrying beer bottles for an after-landing celebration. Payment is 8000 credits."
```

	passengers 5 10

> 	repeat 3

Reviewer: Why the limit?

Author: I moved the limit only to the job with the fine. It doesn't seem right to bother the player with multiple fines (such joke is only funny once).

	job
	source

> 		government "Republic" "Free Worlds" "Syndicate"

Reviewer: ```suggestion
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"
```

	destination
		distance 7 9

> 		government "Republic" "Free Worlds" "Syndicate"

Reviewer: ```suggestion
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"
```

> 	to offer
> 		and
> 			"Spacediving professionals: done" >= 1
> 			"Spacediving amateurs: offered" <= 6
> 			"Spacediving fools: offered" <= 2
> 			credits >= 1000000
> 			random > 40

Reviewer: You don't need the other offer conditions if you're already using `repeat` to limit the offer count.
```suggestion
	to offer
		has "Spacediving professionals: done"
		credits >= 1000000
		random < 60
```

Author: I mostly agree, but the "fools" version should only be offered once the "amateurs" and the "defaulters" option both have been done. The player should have sufficient warning up-front that this mission might be less good than advertised.

	on visit
		dialog phrase "generic passenger on visit"
	on complete

> 		dialog "The skydivers were too affraid to jump and refuse to pay after you land with them. Maybe next time check if your passengers have suitable preparation for a high-altitude jump like this."

Reviewer: Considering that you've brought them a considerable chunk away from their departing planet, I'd expect they'd at least pay compensation.
```suggestion
		dialog "As you descend through the upper atmosphere, you inform your skydivers that it's time to jump. However, after being able to see the distances involved, all of them are far less keen on the idea than when you first picked them up. You bring them down to the landing pad, and they pay you a meagre <payment> before hopping off."
```

Author: They now pay 3000 credits.

mission "Spacediving fools"
	name "Spacediving at <destination>"

> 	description "Drop skydivers from the edge of space. They are carrying cooked chicken wings, fashionable umbrellas and whiskey bottles to ensure the trip will be a grandiose party. Payment is 8000 credits."

Reviewer: ```suggestion
	description "Head to destination with <bunks> skydivers, then drop them from the edge of space as you land. They are carrying cooked chicken wings, fashionable umbrellas and bottles of whiskey for a grand after-landing celebration. Payment is 8000 credits."
```

	passengers 5 10
	repeat 2
	job
	source

> 		government "Republic" "Free Worlds" "Syndicate"

Reviewer: ```suggestion
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"
```

	destination
		distance 7 9

> 		government "Republic" "Free Worlds" "Syndicate"

Reviewer: ```suggestion
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"
```

> 	to offer
> 		and
> 			"Spacediving amateurs: done" >= 1
> 			"Spacediving fools: offered" <= 4
> 			credits >= 1000000
> 			random > 40

Reviewer: ```suggestion
	to offer
		has "Spacediving professionals: done"
		credits >= 1000000
		random < 60
```

	on visit
		dialog phrase "generic passenger on visit"
	on complete
		payment -80000

> 		dialog "The skydivers jump from your ship with their umbrellas opened. All die before landing. Authorities investigate the matter and fine you <payment> for allowing unprepared skydivers to die."

Reviewer: ```suggestion
		dialog "As you descend through the upper atmosphere, the skydivers jump from your ship... with their umbrellas open, and while wearing their ordinary clothes. Authorities fine you <payment> when you land for enabling such a dangerous and deadly act."
```

			label reject
			`	Charles looks surprised for a moment, and then, for the first time you've seen, he smiles. "Thank you, Captain <last>. I wish there were more people like you in the Paradise Worlds."`
			`	You wish the couple the best of luck on Earth, and they slowly make their way to the train station. Presumably, their new apartment is far from here.`

mission "Spacediving professionals"
	name "Spacediving at <planet>"

> 	description "Head to destination with <bunks> skydivers, then drop them from the edge of space as you land. They are wearing thermal spacesuits, durable parachutes and oxygen bottles to ensure their survival. Payment is <payment>. This type of job has risks of non-payment and fines."

Reviewer 2: > This type of job has risks of non-payment and fines

How would you know this in advance? This kind of text amounts to giving the game away immediately.

Author: I want to position the skydiving jobs as "higher payment on average, but also higher risk of non-payment or even fines" and I'm searching for the amount of "reasonable upfront warning" vs the amount of "just find out as you are playing".

Author: Maybe I could focus on "just find out as you are playing" where the "negative" jobs only start showing after 10 "good" jobs.

Author: I choose this last strategy (with the negative jobs starting to show some time after the positive ones).

	job
	source
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"
	destination
		distance 1 7
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"
	to offer
		random < 14
	on visit
		dialog phrase "generic passenger on visit"
	on complete
		payment 8000
		dialog "As you descend through the upper atmosphere, the skydivers pay you <payment> before jumping from your ship, spectacularly diving until they open their parachutes and land safely near the spaceport."

mission "Spacediving defaulters"
	name "Spacediving at <planet>"
	description "Head to destination with <bunks> skydivers, then drop them from the edge of space as you land. They are wearing thermal spacesuits, durable parachutes and oxygen bottles to ensure their survival. Payment is 8000. This type of job has risks of non-payment and fines."
	passengers 5 10
	repeat
	job
	source
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"
	destination
		distance 1 7
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"
	to offer
		has "Spacediving professionals: done"
		random < 1
	on visit
		dialog phrase "generic passenger on visit"
	on complete

> 		dialog "As you descend through the upper atmosphere, the skydivers open the airlocks and jump quickly to avoid paying you. They open their parachutes and land far away from the spaceport."

Reviewer 2: Why wouldn't you have arranged payment in advance? This makes you the player seem like a dummy.

Also, there is no difference in the mission description to suggest that this party might not pay, so you're annoying the player for basically no reason. I wrote a similar mission that seems identical to a normal one but hurts the player (shady passengers double-cross) but for that one, the point was to emphasize the risks of associating with dangerous wanted criminals. I don't see any similar point or lesson being communicated here.

Author: What if I add something like "you report the non-payment to the authorities. That doesn't help you to get paid, but the skydivers might have a much harder time to get off the planet again."

Author: I removed the defaulters and replaced them by the accident and disaster versions.

	on complete
		payment 3000
		dialog "As you descend through the upper atmosphere, you inform your skydivers that it's time to jump. However, after being able to see the distances and atmospheric conditions involved, all of them are far less keen on the idea than when you first picked them up. You bring them down to the landing pad, and they pay you a meagre <payment> before hopping off."

mission "Spacediving fools"
	name "Spacediving at <planet>"
	description "Head to destination with <bunks> skydivers, then drop them from the edge of space as you land. They are carrying fashionable umbrellas and bottles of whiskey for a grand after-landing celebration. Payment is 8000 credits. This type of job has risks of non-payment and fines."
	passengers 5 10
	repeat 20
	job
	source
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"
	destination
		distance 1 7
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"
	to offer
		has "Spacediving defaulters: done"
		has "Spacediving amateurs: done"
		not "Spacediving fools: done"
		credits >= 100000
		random < 10
	on visit
		dialog phrase "generic passenger on visit"
	on complete
		payment -10000

> 		dialog "As you descend through the upper atmosphere, the skydivers jump from your ship... with their umbrellas open, and while wearing their ordinary clothes. Authorities fine you <payment> when you land for enabling such a dangerous and deadly act."

Reviewer 2: My original point still stands. Think about how this would work in real life: if you directly and negligently facilitated the death of 5-10 innocent yet stupid people what's gonna happen? a 10k fine? Or life imprisonment for a decade or more?

I know games aren't real life, but you can't bend the rules of physics or how human societies work too much, or else it breaks the player out of the illusion that the game is crafting.

Author: I removed this version completely.

			label reject
			`	Charles looks surprised for a moment, and then, for the first time you've seen, he smiles. "Thank you, Captain <last>. I wish there were more people like you in the Paradise Worlds."`
			`	You wish the couple the best of luck on Earth, and they slowly make their way to the train station. Presumably, their new apartment is far from here.`

mission "Spacediving professionals"
	name "Spacediving at <planet>"

> 	description "Head to destination with <bunks> skydivers, then drop them from the edge of space as you land. They are wearing thermal spacesuits, durable parachutes and oxygen bottles to ensure their survival. Extreme sports like skydiving typically carry more risks than regular passenger transports. Payment is <payment>."

Reviewer 3: Should this be `Head to <destination>`? I notice that none of these descriptions actually use a destination replacement. If the `<destination>` text replacement is too long, then I think that using `<planet>` would be more appropriate than not mentioning the actual destination at all in the description, or say "the destination" instead of just "destination."

> 	name "Spacediving at <planet>"
> 	description "Head to destination with <bunks> skydivers, then drop them from the edge of space as you land. They are wearing thermal spacesuits, durable parachutes and oxygen bottles to ensure their survival. Extreme sports like skydiving typically carry more risks than regular passenger transports. Payment is <payment>."
> 	passengers 5 10
> 	repeat
> 	job

Reviewer 3: Formatting.
```suggestion
	job
	repeat
	name "Spacediving at <planet>"
	description "Head to destination with <bunks> skydivers, then drop them from the edge of space as you land. They are wearing thermal spacesuits, durable parachutes, and oxygen bottles to ensure their survival. Extreme sports like skydiving typically carry more risks than regular passenger transports. Payment is <payment>."
	passengers 5 10
```

	source
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"
	destination
		distance 1 7
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"

> 		not attributes station
> 		not attributes research

Reviewer 3: ```suggestion
		not attributes "station"
		not attributes "research"
```

	to offer
		random < 14
	on visit
		dialog phrase "generic passenger on visit"
	on complete
		payment 8000
		dialog "As you descend through the upper atmosphere, the skydivers jump from your ship, spectacularly diving until they open their parachutes and land safely near the spaceport. You collect your payment of <payment>."

mission "Spacediving professionals accident"

> 	name "Spacediving at <planet>"
> 	description "Head to destination with <bunks> skydivers, then drop them from the edge of space as you land. They are wearing thermal spacesuits, durable parachutes and oxygen bottles to ensure their survival. Extreme sports like skydiving typically carry more risks than regular passenger transports. Payment is 8000."
> 	passengers 5 10
> 	repeat
> 	job

Reviewer 3: ```suggestion
	job
	repeat
	name "Spacediving at <planet>"
	description "Head to destination with <bunks> skydivers, then drop them from the edge of space as you land. They are wearing thermal spacesuits, durable parachutes, and oxygen bottles to ensure their survival. Extreme sports like skydiving typically carry more risks than regular passenger transports. Payment is 8000."
	passengers 5 10
```

	source
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"
	destination
		distance 1 7
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"

> 		not attributes station
> 		not attributes research

Reviewer 3: ```suggestion
		not attributes "station"
		not attributes "research"
```

	to offer
		"Spacediving professionals: done" > 7
		random < 2
	on visit
		dialog phrase "generic passenger on visit"
	on complete
		dialog "As you descend through the upper atmosphere, the skydivers jump from your ship, spectacularly diving until they try to open their parachutes. Some of the parachutes open only partially and part of the team suffers an uncontrolled hard landing. Medical teams respond quickly, but the skydivers are too preoccupied with their injured teammates to pay you the credits they owe you."

mission "Spacediving professionals disaster"

> 	name "Spacediving at <planet>"
> 	description "Head to destination with <bunks> skydivers, then drop them from the edge of space as you land. They are wearing thermal spacesuits, durable parachutes and oxygen bottles to ensure their survival. Extreme sports like skydiving typically carry more risks than regular passenger transports. Payment is 8000."
> 	passengers 5 10
> 	repeat
> 	job

Reviewer 3: ```suggestion
	job
	repeat
	name "Spacediving at <planet>"
	description "Head to destination with <bunks> skydivers, then drop them from the edge of space as you land. They are wearing thermal spacesuits, durable parachutes, and oxygen bottles to ensure their survival. Extreme sports like skydiving typically carry more risks than regular passenger transports. Payment is 8000."
	passengers 5 10
```

	source
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"
	destination
		distance 1 7
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"

> 		not attributes station
> 		not attributes research

Reviewer 3: ```suggestion
		not attributes "station"
		not attributes "research"
```

	to offer
		has "Spacediving professionals accident: done"
		random < 1
	on visit
		dialog phrase "generic passenger on visit"
	on complete

> 		payment -200000

Reviewer 3: Missions can't complete if you don't have enough credits for a negative payment. I'd suggest checking how many credits the player has in the to offer and ensure that they have enough that they can pay this off, plus a little extra so that a player doesn't lose all their credits from taking this job, or so that the player doesn't have exactly what's needed but falls below that point before reaching the destination; this technically isn't foolproof, but it's at least a safety precaution.

Author: I added a check for 5000k (just a bit more safety margin than that you suggested).
Do you have a reference to the ideas and/or work for the fines?

> 		dialog "As you descend through the upper atmosphere, the skydivers jump from your ship. Some minutes into the jump disaster strikes; the skydivers start moving very spasticly. It's unclear if the spacesuits or the oxygen supplies were at fault, but it looks unlikely that any of the skydivers makes it safely to the ground. Authorities investigate the incident and although they don't find serious wrongdoings on your part, they still fine you <payment> for enabling such a dangerous and deadly act."

Author: The previous warning was very explicit ("carries risk of non-payment and fines"). Maybe a less explicit warning would be appropriate? ("Extreme sports like skydiving typically carry more risks than regular passenger transports.") This doesn't give the game away completely, but still warns the player up-front.

> 		dialog "As you descend through the upper atmosphere, the skydivers jump from your ship. Some minutes into the jump disaster strikes; the skydivers start moving very spasticly. It's unclear if the spacesuits or the oxygen supplies were at fault, but it looks unlikely that any of the skydivers makes it safely to the ground. Authorities investigate the incident and although they don't find serious wrongdoings on your part, they still fine you <payment> for enabling such a dangerous and deadly act."

Reviewer 4: "carries risk of non-payment and fines" is a bit immersion breaking, and what you suggested sounds a lot better. I like it!

mission "Spacediving amateurs"

> 	name "Spacediving at <planet>"
> 	description "Head to destination with <bunks> skydivers, then drop them from the edge of space as you land. They are wearing tricked-out wingsuits, brightly-patterned parachutes and carrying beer bottles for an after-landing celebration. Extreme sports like skydiving typically carry more risks than regular passenger transports. Payment is 8000 credits."
> 	passengers 5 10
> 	repeat
> 	job

Reviewer 3: ```suggestion
	job
	repeat
	name "Spacediving at <planet>"
	description "Head to destination with <bunks> skydivers, then drop them from the edge of space as you land. They are wearing tricked-out wingsuits, brightly-patterned parachutes, and carrying beer bottles for an after-landing celebration. Extreme sports like skydiving typically carry more risks than regular passenger transports. Payment is 8000 credits."
	passengers 5 10
```

	source
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"
	destination
		distance 1 7
		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"
		not attributes station
		not attributes research
	to offer
		"Spacediving professionals: done" > 3
		random < 5
	on visit
		dialog phrase "generic passenger on visit"
	on complete
		payment 30000

> 		dialog "As you descend through the upper atmosphere, you inform your skydivers that it's time to jump. However, after being able to see the distances and atmospheric conditions involved, all of them are far less keen on the idea than when you first picked them up. You bring them down to the landing pad, and they pay you a meagre <payment> before hopping off."

Reviewer 5: ```suggestion
		dialog "As you descend through the upper atmosphere, you inform your skydivers that it's time to jump. However, after being able to see the distances and atmospheric conditions involved, all of them are far less keen on the idea than when you first picked them up. You bring them down to the landing pad, and they pay you a meager <payment> before hopping off."
```
"meagre" would be the British spelling.
